src/java.base/share/classes/java/util/concurrent/LinkedTransferQueue.java

```diff
@@ -865,15 +893,16 @@ final class Itr implements Iterator<E> {
         private void advance(DualNode pred) {
             for (DualNode p = (pred == null) ? head : pred.next, c = p;
                  p != null; ) {
-                final Object item;
-                if ((item = p.item) != null && p.isData) {
+                boolean isData = p.isData;
```

Comment:
@DougLea Are you finding that manual hoisting of reads to final members has a perf edge? 🤔

Reply:
Collection-support code from method countOfMode on down was minimally touched from its previous version. This shows up as a style clash into very branchy self-link-coping code. Changing might require more changes to WhiteBox tests that are finicky about node counts and self-links expected in traversals etc.  (These tests are useful but are themselves hard to change to accommodate small implementation differences.)

Reply:
@DougLea Alright, good to know :)